packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -284,10 +284,15 @@ class ScaleTransition extends AnimatedWidget {
 
   @override
   Widget build(BuildContext context) {
+    /// The ImageFilter layer created by setting filterQuality will introduce
+    /// a saveLayer call. This is usually worthwhile when animating the layer,
+    /// but leaving it in the layer tree before the animation has started or after
+    /// it has finished significantly hurts performance.
+    final bool skipfilterQuality = scale.isCompleted || scale.isDismissed;
```

Comment:
we prefer a switch/case on `Animation.status` over this on the off-chance that we ever need to add another status value to get the analyzer's help for all usages that need to be considered. 

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-if-chains-or--or--with-enum-values

Reply:
Done